homeassistant/components/websocket_api/commands.py

```diff
@@ -923,3 +928,27 @@ async def handle_integration_descriptions(
 ) -> None:
     """Get metadata for all brands and integrations."""
     connection.send_result(msg["id"], await async_get_integration_descriptions(hass))
+
+
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "integration/wait",
+        vol.Required("domain"): str,
+    }
+)
+@decorators.async_response
+async def handle_integration_wait(
+    hass: HomeAssistant, connection: ActiveConnection, msg: dict[str, Any]
+) -> None:
+    """Handle wait for integration command."""
+
+    domain: str = msg["domain"]
+    setup_done = hass.data.get(DATA_SETUP_DONE, {})
+
+    # Wait for the integration to be set up
+    if setup_future := setup_done.get(domain):
+        await setup_future
```

Comment:
Might be nice to have a helper in setup to wait for an integration to be done setup and avoid importing `DATA_SETUP_DONE` here.  We could use it in `config_entries.async_wait_component`

Reply:
A helper is a good idea.
If the dicts in setup.py are not mean for external use they should be protected IMO. I'll do that in a separate PR.